src/Microphysics2M.jl

```diff
@@ -159,36 +150,35 @@ function pdf_rain_parameters(
 end
 
 """
-    size_distribution(pdf, D, q, ρ, N)
+    size_distribution(pdf, q, ρₐ, N)
+    size_distribution(pdf, q, ρₐ, N, D)
 
- - pdf - struct containing size distribution parameters of cloud or rain
- - D - particle size (i.e. maximum dimension of particle)
- - q - cloud or rain water specific content
- - ρₐ - density of air
- - N - cloud or rain water number concentration
+Calculate the size distribution value for rain or cloud particles for a given particle size.
 
-    Returns the size distribution value for rain or cloud particles for a given
-    particle size
+# Arguments
+- `pdf`: Struct containing size distribution parameters of cloud or rain
+- `q`: Cloud or rain water specific content [kg/kg]
+- `ρₐ`: Density of air [kg/m³]
+- `N`: Cloud or rain water number concentration [1/m³]
+- `D`: Particle size (i.e. maximum dimension of particle) [m]
+
+# Returns
+- The first method returns a function that computes the size distribution value
```

Comment:
Could we define the function separately and use that to compute the value? Whats the benefit for having a separate function that returns a function?

Reply:
The primary benefit, as I see it, is that I can create a reference to the size distribution once. In this case, that includes precomputing the line `(; N₀r, λr) = pdf_rain_parameters(pdf, q, ρₐ, N)`. Then I can evaluate it many times during integration by only passing in the diameter. As I see it, it leads to code that more closely resembles the equations from the relevant papers.

We already implicitly do this when we write things like:
```julia
x -> CM2.size_distribution(
    SB2006_no_limiters.pdf_r,
    FT(x),
    qᵣ,
    ρₐ,
    Nᵣ,
)
```
which defines an anonymous function in `x`. The idea with this PR is to make code like that more explicit and reusable.